examples/new/types.ts

```
@@ -0,0 +1,148 @@
+export type ExtractTemplateVariables<T extends string> =
+    T extends `${infer Start}{{${infer Var}}}${infer Rest}`
+        ? Var | ExtractTemplateVariables<Rest>
+        : never;
+
+export type TemplateVariables<T extends string> = Pretty<{
+    [K in ExtractTemplateVariables<T>]: string;
+}>;
+
+export type AgentMemory = {
+    working: WorkingMemory;
+    shortTerm: {};
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)